### DISCUSSION TITLE
How to contribute to EdgeTX Companion?

### DISCUSSION THREAD

**Discussion by micahqcade** [EdgeTX/edgetx#5694] [Q&A]
I would like to contribute to the codebase or contribute via QA. I know there is documentation on how to contribute to EdgeTX, but is there a guide for Companion?

**Comment by pfeerick** [comment]
Not really, no... like most of the codebase it's very much dig in and figure it out affair... since devs generally hate writing documentation, when you're volunteering your time/skills... you get to choose to ignore doing that. 😇  The info at https://github.com/EdgeTX/edgetx/wiki will somewhat relate to building companion also, but that is about all. @elecpower is our resident (er... shacked?) Companion guru... so maybe start up a discussion with him to help get started. And perhaps we can put together some documentation for this in the process 🤭

**Reply by micahqcade** [reply]
Sounds good! I understand that when working for free stuff isn't really documented; maybe I can help somehow with documentation. Its definitely important as I have forgotten how or why I have done things a certain way in the past, and it helps on board devs and QA faster.
